src/Raven.Server/ServerWide/Commands/ETL/UpdateEtlProcessStateCommand.cs

```diff
@@ -99,7 +102,9 @@ protected override BlittableJsonReaderObject GetUpdatedValue(long index, RawData
                 };
             }
 
-            etlState.LastProcessedEtagPerNode[NodeTag] = LastProcessedEtag;
+            if (DbId != null)
+                etlState.LastProcessedEtagPerDbId[DbId] = LastProcessedEtag;
```

Comment:
Shouldn't we also do here
```csharp
if (etlState.LastProcessedEtagPerNode != null)
    etlState.LastProcessedEtagPerNode[NodeTag] = LastProcessedEtag;
```
?

Reply:
added
```
if (etlState.LastProcessedEtagPerNode?.Count > 0)
    etlState.LastProcessedEtagPerNode[NodeTag] = LastProcessedEtag;
```

when Etl State is deserialized from blittable `LastProcessedEtagPerNode` is an empty dictionary, so we don't want to keep updating it.. otherwise the original issue will reside